Lib/asyncio/tasks.py

```diff
@@ -73,15 +73,27 @@ class Task(futures._PyFuture):  # Inherit Python Task implementation
     """A coroutine wrapped in a Future."""
 
     # An important invariant maintained while a Task not done:
+    # _fut_waiter is either None or a Future.  The task can be
+    # in any of 3 states:
     #
-    # - Either _fut_waiter is None, and _step() is scheduled;
-    # - or _fut_waiter is some Future, and _step() is *not* scheduled.
+    # - 1 (_fut_waiter is not None and not _fut_waiter.done());
+    #   __step() is *not* scheduled and the Task is waiting for _fut_waiter.
+    # - 2a (_fut_waiter is None or _fut_waiter.done()), and __wakeup() is scheduled;
+    #   the Task is waiting for __wakeup() to be executed.
+    # - 2b (_fut_waiter is None or _fut_waiter.done()), and __step() is scheduled;
+    #   the Task is waiting for __step() to be executed.
+    # - 3 _fut_waiter is None and __step() is *not* scheduled;
+    #   the Task is currently executing (in __step()).
```

Comment:
I can't speak to the content, but a formatting suggesion for readability:
```suggestion
    # - 1:  _fut_waiter is not None and not _fut_waiter.done()):
    #       __step() is *not* scheduled and the Task is waiting for _fut_waiter.
    # - 2a: _fut_waiter is None or _fut_waiter.done()) and __wakeup() is scheduled:
    #       the Task is waiting for __wakeup() to be executed.
    # - 2b: _fut_waiter is None or _fut_waiter.done()) and __step() is scheduled
    #       the Task is waiting for __step() to be executed.
    # - 3:  _fut_waiter is None and __step() is *not* scheduled:
    #       the Task is currently executing (in __step()).
```

Reply:
Much appreciated :)

Reply:
Yeah, please apply this. I had the same confusion.